device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/twin/TwinProperties.java

```diff
@@ -215,12 +207,7 @@ JsonElement toJsonElementWithMetadata()
      */
     public TwinCollection getDesired()
     {
-        /* SRS_TWIN_PROPERTIES_21_009: [The getDesired shall return a TwinCollection with the stored desired property.] */
-        if (this.desired == null)
-        {
-            return null;
-        }
-        return new TwinCollection(this.desired);
```

Comment:
By returning a clone of the desired properties twin collection instead of the local reference, we could not update them in place. See the modified e2e test to demonstrate what used to not work